src/main/java/com/uwetrottmann/tmdb2/entities/DiscoverFilter.java

```diff
@@ -22,7 +22,13 @@ public DiscoverFilter(Integer... items) {
         this.items = items;
     }
 
+    public DiscoverFilter(Separator separator, Integer... items) {
+        this.separator = separator;
+        this.items = items;
+    }
+
     public DiscoverFilter(Separator separator, ReleaseType... types) {
+        assert types != null;
```

Comment:
Please change to a condition and throw an exception instead. Depending on the platform assert may not work. Same for the assert usage below.

Reply:
Is the following changes ok ?
```
if (types == null){
            throw new IllegalArgumentException();
}
```
```
if (separator == null){
            throw new NullPointerException();
}
```
Or are there any message conventions I should follow ?

Reply:
The platform itself would throw NPE. If you really want the code to throw here explicitly, please throw `IllegalArgumentException` or `IllegalStateException` and use a message why, e.g. `types must not be null`.

Reply:
I really wanted to do a good job, if you think that's unnecessary you can ignore the last commit.
I'm not far from a beginner in Java so thanks for your feedback.

Reply:
About the `separator` being null which is not an argument in this case. I didn't find any definitive answer :
https://stackoverflow.com/questions/271526/avoiding-nullpointerexception-in-java
https://stackoverflow.com/questions/3322638/is-it-okay-to-throw-nullpointerexception-programmatically
**Annotations** beforehand in constructors, **Asserts**, **checkNotNull()**, **NPE**, **nothing**...
 Choose your school of though I will comply.